tests/kohana/UTF8Test.php

```diff
@@ -359,9 +359,18 @@ public function provider_strcasecmp()
 	 * @test
 	 * @dataProvider provider_strcasecmp
 	 */
-	public function test_strcasecmp($input, $input2, $expected)
+	public function test_strcasecmp(string $input, string $input2, int $expected)
 	{
-		$this->assertSame($expected, UTF8::strcasecmp($input, $input2));
+		$actual = UTF8::strcasecmp($input, $input2);
+
+		if ($expected === 0){
+			$this->assertSame($expected, $actual);
+		} elseif( $expected < 0 ){
```

Comment:
Per Slack, I think let's just change the behaviour/expectation so that it is strictly always `-1/1/0` across all supported versions. I don't think anything should be actually using the result other than as a `less/equal/greater` flag, we can note in the changelog.